plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -256,330 +133,154 @@ def sequence_files(
-                            return env.from_string(in_str, globals=globals)
-
-            return old_from_string(*args, **kwargs)
-
-        node = self._find_node(fname, config)
-        templater_logger.debug(
-            "_find_node for path %r returned object of type %s.", fname, type(node)
-        )
-
-        save_ephemeral_nodes = dict(
-            (k, v)
-            for k, v in self.dbt_manifest.nodes.items()
-            if v.config.materialized == "ephemeral"
-            and not getattr(v, "compiled", False)
+    def _unsafe_process(
+        self, fname: Optional[str], in_str: str, config: FluffConfig = None
+    ):
+        # Get project
+        # osmosis_dbt_project = self.dbt_project_container.get_project_by_root_dir(
+        #     self.project_dir
+        # )
+        osmosis_dbt_project = self.dbt_project_container.get_project_by_root_dir(
+            # from .sqlfluff templater project_dir
+            config.get_section((self.templater_selector, self.name, "project_dir"))
         )
-        with self.connection():
-            # Apply the monkeypatch.
-            Environment.from_string = from_string
-            try:
-                node = self.dbt_compiler.compile_node(
-                    node=node,
-                    manifest=self.dbt_manifest,
-                )
-            except Exception as err:
-                templater_logger.exception(
-                    "Fatal dbt compilation error on %s. This occurs most often "
-                    "during incorrect sorting of ephemeral models before linting. "
-                    "Please report this error on github at "
-                    "https://github.com/sqlfluff/sqlfluff/issues, including "
-                    "both the raw and compiled sql for the model affected.",
-                    fname,
-                )
-                # Additional error logging in case we get a fatal dbt error.
-                raise SQLFluffSkipFile(  # pragma: no cover
-                    f"Skipped file {fname} because dbt raised a fatal "
-                    f"exception during compilation: {err!s}"
-                ) from err
-            finally:
-                # Undo the monkeypatch.
-                Environment.from_string = old_from_string
-
-            if hasattr(node, "injected_sql"):
-                # If injected SQL is present, it contains a better picture
-                # of what will actually hit the database (e.g. with tests).
-                # However it's not always present.
-                compiled_sql = node.injected_sql
-            else:
-                compiled_sql = getattr(node, COMPILED_SQL_ATTRIBUTE)
+        if not osmosis_dbt_project:
+            osmosis_dbt_project = self.dbt_project_container.add_project(
+                # name_override=x_dbt_project,
+                project_dir=self.project_dir,
+                profiles_dir=self.profiles_dir,
+                # target=target,
+                vars=self._get_cli_vars(),
```

Comment:
The `vars` parameter won't be allowed until https://github.com/z3z1ma/dbt-osmosis/pull/30 is merged and released.